File: LightsOutPuzzle.Common/Tile.cs
namespace LightsOutPuzzle.Common
{
    public class Tile
    {
        public Tile(int row, int column, bool imageVisible = false)
        {
            Row = row;
            Column = column;
            ImageVisible = imageVisible;
            PreviousState = imageVisible;
        }
        public int Column { get; set; }
        public int Row { get; set; }
        public bool ImageVisible { get; set; }
        public bool PreviousState { get; set; }
        public bool HasBeenFlipped => PreviousState != ImageVisible;
        public void Flip()
        {
            PreviousState = ImageVisible;
            ImageVisible = !ImageVisible;
        }
    }
}

File: LightsOutPuzzle.Flutter/lib/model/tile.dart
class Tile {
  int _row;
  int _column;

  int get row => _row;
  int get column => _column;
  bool imageVisible;
  bool previousState;
  bool get hasBeenFlipped => previousState != imageVisible;

  Tile(this._row, this._column, {visible = false})
      : this.imageVisible = visible,
        this.previousState = visible;

  void flip() {
    previousState = imageVisible;
    imageVisible = !imageVisible;
  }

  Tile clone() {
    return Tile(_row, _column, visible: imageVisible);
  }
}
